scripts/network/rpcs/gameserver/sync.gd

```diff
@@ -395,3 +398,72 @@ func playersynchronizer_sync_skill_use(p: Vector2, s: String):
 
 	if user.player.component_list.has("player_synchronizer"):
 		user.player.component_list["player_synchronizer"].sync_skill_use(p, s)
+
+
+#Only client can make this RPC, runs on server
+@rpc("call_remote", "any_peer", "reliable")
+func characterclasscomponent_sync_all(n: String):
+	if not G.is_server():
+		return
+
+	var id: int = multiplayer.get_remote_sender_id()
+
+	# Only allow logged in players
+	if not G.is_user_logged_in(id):
+		return
+
+	var entity: Node = G.world.get_entity_by_name(n)
+
+	if entity == null:
+		return
+
+	if entity.get("component_list") == null:
+		return
+
+	if entity.component_list.has("class_component"):
+		entity.component_list["class_component"].sync_all(id)
+
+
+#Only server can make this RPC, runs on client
+@rpc("call_remote", "authority", "reliable")
+func characterclasscomponent_sync_response(n: String, d: Dictionary):
+	assert(not G.is_server(), "This method is only intended for client use")
+	var entity: Node = G.world.get_entity_by_name(n)
+
+	if entity == null:
+		return
+
+	if entity.get("component_list") == null:
+		return
+
+	if entity.component_list.has("class_component"):
+		entity.component_list["class_component"].sync_response(d)
+
+
+#Only client can make this RPC, runs on server
+@rpc("call_remote", "any_peer", "reliable")
+func characterclasscomponent_sync_class_change(n: String, c: Array[String]):
+	if not G.is_server():
+		return
+
+	var id: int = multiplayer.get_remote_sender_id()
+
+	# Only allow logged in players
+	if not G.is_user_logged_in(id):
+		return
+
+	var entity: Node = G.world.get_entity_by_name(n)
+
+	if entity == null:
+		return
+
+	if entity.get("component_list") == null:
+		return
+
+	if entity.component_list.has("class_component"):
+		#The component can run it's own checks to verify it is allowed to change them.
+		entity.component_list["class_component"].replace_classes(c)
```

Comment:
```suggestion
func characterclasscomponent_sync_class_change(n: String, c: Array):
	if not G.is_server():
		return

	var id: int = multiplayer.get_remote_sender_id()

	# Only allow logged in players
	if not G.is_user_logged_in(id):
		return

	var entity: Node = G.world.get_entity_by_name(n)

	if entity == null:
		return

	if entity.get("component_list") == null:
		return

	if entity.component_list.has("class_component"):
		#The component can run it's own checks to verify it is allowed to change them.

		#Temporary adding this typecast to prevent errorcode caused by: https://github.com/godotengine/godot/issues/69215
		var stringArray: Array[String] = []
		stringArray.assign(c)
		entity.component_list["class_component"].replace_classes(stringArray)
```